### PR TITLE
Set up release machinery for v0.1.0

### LINKED ISSUE #64: Set up release machinery and cut v0.1.0
## Background

Thrust has 185 commits, zero git tags ever, and `Cargo.toml` stuck at `version = "0.1.0"` since day one. The work is past the point where a tagged release is the natural next step:

- **Algorithms**: PPO (clipped + GAE) and now DQN (vanilla MVP, landed via #52).
- **Environments**: CartPole, Snake (single + multi-agent), Pong (single + self-play via #53), Bucket-Brigade (external).
- **Multi-agent infra**: `Population`, `GameSimulator`, `PolicyLearner`, four matchmaking strategies, `JointMultiAgentTrainer`.
- **Tooling**: WASM inference for browser demos, libtorch GPU support, Bayesian + Pareto hyperparameter optimization.
- **Docs**: `ROADMAP.md`, `MULTI_AGENT_DESIGN.md`, `docs/PPO_BEST_PRACTICES.md`, `docs/RESEARCH_PAPERS.md`, `docs/RL_TOYBOX_SURVEY.md` (landed via #49).

**Crates.io status (verified 2026-06-05):** `thrust-rl`, `thrust_rl`, and plain `thrust` are all unclaimed. We are NOT on crates.io despite the framework being mature enough to warrant it. Part of this issue is to fix that before someone squats the name.

## What this issue does

Adapt the Loom-installer-shipped release skill into a thrust-specific one, draft a first `CHANGELOG.md`, cut v0.1.0, and publish to crates.io.

### Sub-tasks

1. **Replace `.claude/commands/loom/release.md`** — it currently has Loom-framework verbatim content (references `loom-daemon`, `loom-api`, `mcp-loom/package.json`, "5 version-bearing files" that don't exist here). Either:
   - (a) Rewrite it for Thrust: version-bearing file is the root `Cargo.toml`. Possibly also a version constant in `src/lib.rs` if introduced. The release workflow's job list includes `cargo publish` to crates.io.
   - (b) Delete it entirely and document the release procedure in `CONTRIBUTING.md` or a `docs/RELEASING.md` (simpler if `/loom:release` is not going to be the canonical path here).
   
   **Prefer (a)** so `/loom:release` becomes idiomatic for thrust over time.

2. **Create `CHANGELOG.md`** at the repo root following Keep-a-Changelog conventions. The v0.1.0 entry can be generated from `git log --oneline` and curated down to user-visible categories:
   - Algorithms (PPO, DQN)
   - Environments (CartPole, Snake, Pong, Bucket-Brigade)
   - Multi-agent infra
   - WASM / web demos
   - Hyperparameter optimization
   - Docs
   - Known limitations (DQN doesn't reliably solve CartPole — tracked by #58; etc.)

3. **Document the release procedure** in `CHANGELOG.md` or `docs/RELEASING.md`:
   - When to bump major / minor / patch (semver, but pre-1.0 means breaking changes go to minor not major)
   - How to draft release notes
   - The literal commands to tag, push, and publish (including `cargo publish` and what needs to be in `Cargo.toml` for the publish to succeed: `description`, `license`, `repository`, `readme`, `keywords`, `categories`)
   - How to create the GitHub Release

4. **Prepare `Cargo.toml` for crates.io publishing** — verify and add as needed:
   - `description` (required)
   - `license` (required — likely MIT or Apache-2.0 or both; check existing LICENSE file)
   - `repository` (should be `https://github.com/rjwalters/thrust`)
   - `readme` (point at `README.md`)
   - `keywords` (≤5, ≤20 chars each, alphanumeric + `-`/`_`)
   - `categories` (from the standard list — `science::ml`, `algorithms` are likely fits)
   - `homepage` (optional, point at the GitHub Pages demo if appropriate)
   - `documentation` (optional, will default to docs.rs)
   - Confirm no `publish = false` flag is present.

5. **Publishing logistics**:
   - Run `cargo publish --dry-run` first to validate the manifest. Fix anything it flags.
   - The `train_*` binary targets in `Cargo.toml` should NOT prevent publishing as a library. If they do, gate them behind a feature or `required-features` so the crate's default profile is library-only.
   - Confirm `cargo package --list` doesn't include unwanted files (build artifacts, large datasets). May need to add `exclude = [...]` or use `include = [...]` in `[package]`.
   - The actual `cargo publish` should be done by the maintainer (rjwalters) with their crates.io token — NOT by an autonomous agent. The PR delivers a publish-ready state; the maintainer executes the final `cargo publish` command from a green main after merge.

6. **Cut v0.1.0**:
   - `git tag v0.1.0 <sha>` on a commit that includes the new CHANGELOG.md and Cargo.toml metadata
   - `git push origin v0.1.0`
   - `gh release create v0.1.0 --title "v0.1.0" --notes-file <(awk '/^## \[0.1.0\]/{f=1;next} /^## \[/{f=0} f' CHANGELOG.md)`
   - Maintainer runs `cargo publish` from main at the tagged commit.

## Out of scope (file separate issues if/when needed)

- A `release.yml` GitHub Actions workflow that builds binary artifacts. **Decision: no binary build for v0.1.0.** GitHub Release will pull notes from `CHANGELOG.md`; consumers build from `cargo` if they want the code.
- Automating `cargo publish` from CI. **Decision: maintainer publishes manually** for v0.1.0 — first publish is too important to delegate to CI. Automate later once the procedure is settled.
- Bumping `web/package.json`'s version. The web subproject doesn't ship as an npm package and isn't versioned today (`version: "0.0.0"`). Leave it alone unless someone needs a versioned npm artifact.

## Acceptance criteria

- [ ] `.claude/commands/loom/release.md` either rewritten for Thrust or removed (with a replacement procedure documented elsewhere).
- [ ] `CHANGELOG.md` exists at root with a non-trivial `## [0.1.0]` section.
- [ ] Release procedure documented somewhere discoverable (CHANGELOG.md preamble or `docs/RELEASING.md`).
- [ ] `Cargo.toml` has all crates.io-required metadata (`description`, `license`, `repository`, `readme`) and validated `cargo publish --dry-run`.
- [ ] `v0.1.0` tag pushed to `origin`.
- [ ] GitHub Release for v0.1.0 created and visible.
- [ ] **(Maintainer step)** `cargo publish` succeeds; `thrust-rl v0.1.0` is live on crates.io and docs.rs.

## Notes for the curator / builder

- This is mostly a documentation + metadata PR. No code changes to crate logic.
- The PR should leave the repo in a "ready to publish" state. The actual `cargo publish` is a maintainer post-merge step.
- DO NOT attempt to publish to crates.io from a builder agent — that requires a crates.io API token that should not be in agent reach.
- After v0.1.0 lands, `Cargo.toml`'s version should bump to `0.2.0-dev` (or similar — confirm with the maintainer's convention) so the unreleased `main` branch is distinguishable from the released tag. This may be a follow-up.

---

## Curator note (2026-06-08)

Confirmed the issue is ready for a Builder. Concrete findings from a fresh inspection of the repo:

**LICENSE files present:** `LICENSE-APACHE` and `LICENSE-MIT` both exist at the repo root. The dual-license posture is already real on disk, so `Cargo.toml`'s existing `license = "MIT OR Apache-2.0"` is correct as-is and does not need to change. No new LICENSE file needs to be added.

**`Cargo.toml` metadata — gap is placeholders, not missing fields.** Most crates.io-required fields are already declared but several contain default-template placeholders that will cause an incorrect or rejected publish:
- `authors = ["Your Name <your.email@example.com>"]` — placeholder; replace with the real maintainer attribution (or remove; `authors` is optional on recent Cargo).
- `repository = "https://github.com/yourusername/thrust"` — placeholder; should be `https://github.com/rjwalters/thrust`.
- `categories = ["science", "algorithms"]` — `"science"` is a valid top-level crates.io category but `"science::ml"` is more specific and likely a better fit; verify against the [official category list](https://crates.io/category_slugs) and pick the most accurate slugs.
- `description`, `keywords`, `readme`, `license` — already populated and reasonable; no change required.
- No `homepage` or `documentation` keys today — optional, but the GitHub Pages demo URL would be a natural `homepage`.
- No `publish = false` flag — good.

**`.claude/commands/loom/release.md` — confirmed Loom-verbatim and not actionable for Thrust.** Current contents reference `./scripts/version.sh` (does not exist here), "5 version-bearing files" enumerated as `package.json`, `mcp-loom/package.json`, `loom-daemon/Cargo.toml`, `loom-api/Cargo.toml`, `CLAUDE.md` (none of these are version-bearing in Thrust except a hypothetical root `Cargo.toml`), and a `release.yml` GitHub Actions workflow that does not exist in this repo. The Builder should rewrite from scratch per option (a) in the issue; trying to in-place edit will be more confusing than a clean rewrite.

**`CHANGELOG.md` — confirmed does not exist at repo root.** Builder creates from scratch; no merge to worry about.

**Examples target gating — already correct.** Every `[[example]]` entry in `Cargo.toml` already declares `required-features = ["training"]` (or `["training", "env-bucket-brigade"]` for the Bucket Brigade target). The acceptance criterion about not blocking library publication should be satisfied as written, but the Builder should still run `cargo publish --dry-run` to confirm.

No body rewrite is needed beyond this note. Acceptance criteria are sound.

### PR DESCRIPTION
**Pull request by rjwalters**
Closes #64.

## Summary

Adds the crate-level metadata, documentation, and tooling needed to publish `thrust-rl` to crates.io as v0.1.0. The actual `cargo publish`, `git tag v0.1.0`, and GitHub Release creation are explicit maintainer post-merge steps, per the issue body:

> The actual `cargo publish` should be done by the maintainer (rjwalters) with their crates.io token --- NOT by an autonomous agent. The PR delivers a publish-ready state; the maintainer executes the final `cargo publish` command from a green main after merge.

### `Cargo.toml`

- Replaced placeholder `authors` and `repository` with real values (Robb Walters, `rjwalters/thrust`).
- Expanded `description`. Refined `keywords` (5 entries, all <=20 chars, valid alphanumeric/`-`/`_`): `reinforcement`, `machine-learning`, `rl`, `ppo`, `dqn`.
- Tightened `categories` to entries from the [official crates.io slug list](https://crates.io/category_slugs): `science`, `algorithms`, `game-development`, `simulation`.
- Added `homepage` (the GitHub Pages demo) and `documentation` (docs.rs) keys.
- Added a comprehensive `exclude = [...]` block so the published tarball drops model checkpoints (`*.pt`, `*.safetensors`, `models/**`), the web demo, the `envs/bucket-brigade/` submodule, training scripts, `.loom/`, `.claude/`, `.github/`, and project planning docs.
- **Disabled the `env-bucket-brigade` feature** for the published manifest. `cargo publish` rejects path-only deps without a version, and `bucket-brigade-core` is not yet on crates.io. The dep and feature lines are commented out (not deleted); local builds can un-comment two blocks in `Cargo.toml` to use the env. The source files under `src/env/games/bucket_brigade/` remain in-tree (they are `#[cfg(feature = "env-bucket-brigade")]` gated, so the published crate simply omits them). Tracked as a v0.2.x follow-up.

### `CHANGELOG.md` (new, root)

Keep-a-Changelog format with an `[Unreleased]` placeholder and a fully populated `[0.1.0] - 2026-06-08` entry. The 0.1.0 entry curates `git log --oneline` (191 commits since project start) into the user-visible categories the issue called out:

- Algorithms (PPO clipped + GAE + aux-loss hook; DQN, Double-DQN with Polyak soft updates, Prioritized Experience Replay).
- Environments (CartPole, Snake single+multi, Pong single+self-play, SimpleBandit, ContinuousLqr, BucketBrigade source-only).
- Multi-agent infra (`Population`, `JointMultiAgentTrainer`, four matchmakers, `clone_state`/`restore_state`).
- WASM and browser demos.
- Hyperparameter optimization.
- Documentation.
- Tooling.
- **Known limitations**: DQN doesn't reliably solve CartPole; `env-bucket-brigade` is source-only in the published crate; `multi_agent::PolicyLearner` API is experimental; `tch` 0.22 + Rust nightly + edition 2024; no binary release artifacts.

### `docs/RELEASING.md` (new)

Maintainer-facing release procedure documenting:

- Pre-1.0 semver convention (breaking changes go to MINOR until `1.0.0`).
- Pre-flight checks (CI green, no open PRs, clean tree).
- `CHANGELOG.md` and `Cargo.toml` update steps (with literal commands).
- `cargo publish --dry-run` validation, both with and without libtorch.
- `cargo package --list` inspection.
- The literal `git tag -a vX.Y.Z`, `git push origin vX.Y.Z`, `gh release create` commands.
- `cargo publish` hand-off (token-bearing, maintainer-only).
- Yanking a bad release.
- Explicit out-of-scope items (CI-driven publish, binary artifacts, workspace publishing, `web/package.json`).

### `.claude/commands/loom/release.md` (rewritten)

The original file was Loom-framework verbatim --- referenced `loom-daemon`, `loom-api`, `mcp-loom/package.json`, a `scripts/version.sh` that doesn't exist here, and "5 version-bearing files" that don't exist either. Replaced with a 9-phase Thrust-specific interactive walkthrough: pre-flight, gather changes, semver decision (pre-1.0 calibrated), draft CHANGELOG, apply changes, commit/tag/push, create GitHub Release, hand off to maintainer for `cargo publish`, post-release summary. Explicit note that `cargo publish` must not be run from an autonomous agent (token exposure).

## Acceptance criteria coverage

From the issue body:

- [x] `.claude/commands/loom/release.md` rewritten for Thrust.
- [x] `CHANGELOG.md` exists at root with a non-trivial `## [0.1.0]` section.
- [x] Release procedure documented in `docs/RELEASING.md`.
- [x] `Cargo.toml` has all crates.io-required metadata (`description`, `license`, `repository`, `readme`) and validated `cargo publish --dry-run`.
- [ ] **(Maintainer step)** `v0.1.0` tag pushed to `origin`.
- [ ] **(Maintainer step)** GitHub Release for v0.1.0 created and visible.
- [ ] **(Maintainer step)** `cargo publish` succeeds; `thrust-rl v0.1.0` is live on crates.io and docs.rs.

The three unchecked items are the maintainer's responsibility per the explicit "Publishing logistics" guidance in the issue.

## Validation performed

- `cargo publish --dry-run --no-default-features --allow-dirty`: **OK**
- `LIBTORCH_USE_PYTORCH=1 cargo publish --dry-run --allow-dirty`: **OK** (uses host pytorch install for libtorch).
- `cargo test --lib --features training`: **207/207 passing.**
- `cargo test --tests --features training`: **all targets passing.**
- `cargo check --no-default-features`: **OK** (10 pre-existing warnings, unchanged by this PR).
- `cargo package --list`: 143 files, 316 KB tarball. Confirmed excluded: model checkpoints, `web/`, `web-old/`, `envs/**`, `scripts/**`, `.loom/`, `.claude/`, `.github/`, optimization result JSON.

## Maintainer post-merge checklist

From a clean checkout of `main` after merge:

```bash
# 1. Tag
git checkout main && git pull
git tag -a v0.1.0 -m "thrust-rl v0.1.0"
git push origin v0.1.0

# 2. GitHub Release with notes from CHANGELOG
gh release create v0.1.0 \
    --title "v0.1.0" \
    --notes-file <(awk '/^## \[0\.1\.0\]/{f=1;next} /^## \[/{f=0} f' CHANGELOG.md)

# 3. Publish to crates.io (token required, do NOT delegate to agents)
LIBTORCH_USE_PYTORCH=1 cargo publish
# or
cargo publish --no-verify
```

Verify within a few minutes:
- `https://crates.io/crates/thrust-rl/0.1.0` is live.
- `https://docs.rs/thrust-rl/0.1.0/` is building.

## Test plan

- [ ] `cargo publish --dry-run --no-default-features` succeeds on CI.
- [ ] `cargo publish --dry-run` succeeds with libtorch present on CI.
- [ ] All existing tests still pass (`cargo test --features training`).
- [ ] `cargo check --no-default-features` still compiles.
- [ ] `cargo package --list` output reviewed: no model files, no submodule, no web demo.

### CODE REVIEW

**Comment by rjwalters** [comment]
LGTM. Approved.

## Judge review summary

This is a clean, well-scoped release-machinery PR. The diff is exclusively metadata, manifest, and docs — no library logic touched. Verified locally:

- `cargo publish --dry-run --no-default-features --allow-dirty` succeeds.
- `cargo package --list` shows 143 files. Spot-checked: no `*.pt`/`*.safetensors`, no `web/`, no `web-old/`, no `envs/`, no `.loom/`, no `.claude/`, no `.github/`, no `scripts/`.
- `Cargo.toml` metadata is complete and crates.io-valid: `description`, `license` (`MIT OR Apache-2.0`), `repository`, `homepage`, `documentation`, `readme`, `keywords` (5 entries, all <=20 chars, alphanumeric/`-`/`_`), `categories` (all from the official crates.io slug list).
- `CHANGELOG.md` follows Keep a Changelog, has a non-trivial curated `## [0.1.0] - 2026-06-08` section grouped under Added / Known limitations / Removed, plus the `[Unreleased]` placeholder and link refs.
- `docs/RELEASING.md` is a thorough maintainer-facing procedure with the literal `cargo publish` / `git tag` / `gh release create` commands and an explicit out-of-scope section.
- `.claude/commands/loom/release.md` was rewritten (not deleted) into a 9-phase Thrust-specific interactive walkthrough; no leftover references to `loom-daemon`, `mcp-loom`, or `scripts/version.sh`.
- No `v0.*` tag exists locally and no GitHub release exists yet — the maintainer post-merge steps remain pending as the issue requires.
- All 7 CI checks green; mergeStateStatus CLEAN.

## Specific scrutiny: `env-bucket-brigade` trade-off

The Builder handled this correctly. The feature is **commented out, not deleted**:

- `bucket-brigade-core` dep line is commented out with a 9-line explanatory block citing `cargo publish`'s path-only-dep restriction and pointing at `docs/RELEASING.md` and the v0.2.x follow-up.
- `env-bucket-brigade = ["bucket-brigade-core"]` feature line is commented out with a matching explanation.
- The `[[example]] train_p3` block is commented out (not deleted) with the same rationale.
- The `bucket_brigade` source module remains in-tree under `src/env/games/bucket_brigade/` (already `#[cfg(feature = "env-bucket-brigade")]` gated from #73), so the published tarball simply omits compilation of that branch.
- The follow-up commit `d61ed1e` adds a `[lints.rust] unexpected_cfgs` allow-list for the disabled feature name to keep `cargo doc` clean under `-D warnings`, with an inline comment instructing future maintainers to remove the stanza when the feature is re-enabled.
- The trade-off is documented in three places: (a) the inline Cargo.toml comments, (b) the CHANGELOG `### Added > Environments > BucketBrigade` bullet, and (c) a dedicated CHANGELOG `### Known limitations` entry that explicitly states "Re-publishing `bucket-brigade-core` is tracked as a v0.2.x follow-up."

The work from PR #73 is preserved (source stays, gate stays), users on git checkouts can re-enable by uncommenting two blocks, and the v0.2.x intent is recorded. This is the acceptable outcome.

Marking `loom:pr` for the maintainer to merge and then execute the explicit post-merge tag / release / publish steps documented in the PR body and `docs/RELEASING.md`.
